improve efficient training on CPU documentation (#28646)
huggingface/transformers: docs/source/en/perf_train_cpu.md

This guide focuses on training large models efficiently on CPU.

## Mixed precision with IPEX

> Mixed precision is the use of both single (FP32) and half-precision (BF16/FP16) data types in a model to accelerate training or inference while still preserving much of the single-precision accuracy. Modern CPUs such as 3rd Gen Intel® Xeon® Scalable processors and 4th Gen Intel® Xeon® Scalable processors support the lower-precision data type BFloat16 natively, so you should get performance benefits out of the box by simply turning the mixed precision training with BFloat16 on.

Reviewer: To make things a little more concise :)

```suggestion
Mixed precision uses single (fp32) and half-precision (bf16/fp16) data types in a model to accelerate training or inference while still preserving much of the single-precision accuracy. Modern CPUs such as 3rd and 4th Gen Intel® Xeon® Scalable processors natively support bf16, so you should get more performance out of the box by enabling mixed precision training with bf16.
```

> To further maximize training performance, you can use Intel® Extension for PyTorch (IPEX), which is a library built upon PyTorch and adds additional CPU ISA level support such as Intel® Advanced Vector Extensions 512 (Intel® AVX-512) Vector Neural Network Instructions (VNNI) and Intel® Advanced Matrix Extensions (Intel® AMX) for an extra performance boost on Intel CPUs. However, it is not guaranteed that CPUs with only AVX2 (e.g., AMD CPUs or older Intel CPUs) have better performance under IPEX.

Reviewer: ```suggestion
To further maximize training performance, you can use Intel® Extension for PyTorch (IPEX), which is a library built on PyTorch and adds additional CPU instruction level architecture (ISA) level support such as Intel® Advanced Vector Extensions 512 (Intel® AVX-512), Vector Neural Network Instructions (VNNI), and Intel® Advanced Matrix Extensions (Intel® AMX) for an extra performance boost on Intel CPUs. However, CPUs with only AVX2 (e.g., AMD or older Intel CPUs) are not guaranteed to have better performance under IPEX.
```

Reviewer 2: "Intel® Advanced Vector Extensions 512 (Intel® AVX-512) Vector Neural Network Instructions (VNNI)" should be "Intel® Advanced Vector Extensions 512 Vector Neural Network Instructions (Intel® AVX512-VNNI)" VNNI is a vector neural network instruction, which is a subset of AVX512 ISA. Thx @stevhliu 's help.

> The Auto Mixed Precision for CPU backend has been enabled since PyTorch-1.10. At the same time, the support of Auto Mixed Precision with BFloat16 for CPU and BFloat16 optimization of operators has been massively enabled in Intel® Extension for PyTorch, and partially upstreamed to PyTorch master branch. Users can get better performance and user experience with IPEX Auto Mixed Precision.

Reviewer: ```suggestion
Auto Mixed Precision (AMP) for CPU backends has been enabled since PyTorch 1.10. AMP support for bf16 on CPUs and bf16 operator optimization is also supported in IPEX and partially upstreamed to the main PyTorch branch. You can get better performance and user experience with IPEX AMP.
```
